packages/server/src/modules/user/service.ts

```diff
@@ -4,8 +4,23 @@ import { AppDocument } from '@types';
 import { compare, hash } from 'bcrypt';
 import { FilterQuery } from 'mongoose';
 import { utils, service, db } from '@application';
+import { shuffle } from 'lodash';
 import { UserModel } from './model';
 
+const PUBLIC_COACH_NUMBER = 2;
+
+const checkCoachPublicInfo = (
```

Comment:
checkX in this case isn't a good name. Should be `withCoachPublicInfo` since you're decorating the query state in a way.